### PR TITLE
new package: vnstat

### PR DESCRIPTION
**Pull request by Denzy7**
vnstatd needs to be run as root since it reads /sys/class/net. vnstat also needs root since it reads databse from vnstatd. vnstati pulls in libgd as dependency which depends on x11 libs (can be added as seperate package?)

### CODE REVIEW

**Comment by TomJo2000** [comment]
I'd assume `vnstati` is no good without `vnstatd`, so it might make sense to make it sub-package.
But it'd also have to go into the Root repo.

**Comment by Denzy7** [comment]
Yep. It's even how [debian does it](https://packages.debian.org/sid/net/vnstati)

File: root-packages/vnstat/build.sh
Comment (TomJo2000) [inline root-packages/vnstat/build.sh:8]: We'll probably also want a `TERMUX_PKG_SERVICE_SCRIPT` for a `sv` service for the daemon.

Should be able to interpolate from the upstream Systemd service example.
https://github.com/vergoh/vnstat/blob/19bacde8d7123c4184fcc4b2926aab0e1ada2d6d/examples/systemd/simple/vnstat.service#L6-L9

Comment (Denzy7) [reply root-packages/vnstat/build.sh:8]: Is `sv` run as root?

**Comment by TomJo2000** [comment]
> Yep. It's even how [debian does it](https://packages.debian.org/sid/net/vnstati). Not sure how the package can be split since vnstati source is also in vnstat repo. Perhaps a custom repo with the source Idk

Subpackages work very similarly to regular `build.sh` files, but are a bit simpler.
https://github.com/termux/termux-packages/wiki/Creating-new-package#writing-a-subpackage-script

Here's a simple example from fcitx5
https://github.com/termux/termux-packages/blob/970362f18c64c8c4a3340db70dbeca77f167e720/x11-packages/fcitx5-gtk-common/fcitx5-gtk4.subpackage.sh#L1-L3

File: root-packages/vnstat/build.sh
Comment (TomJo2000) [reply root-packages/vnstat/build.sh:8]: No its run entirely in userspace.
It should suffice to set:
```suggestion
TERMUX_PKG_DEPENDS="libsqlite"
TERMUX_PKG_SERVICE_SCRIPT=('vnstatd' 'vnstatd -n')
```
The format is: TERMUX_PKG_SERVICE_SCRIPT=(`daemon-name` `daemon-command`)

Comment (Denzy7) [reply root-packages/vnstat/build.sh:8]: Nice. I guess I'll also have to copy [how docker cleans up after sv down](https://github.com/termux/termux-packages/blob/master/root-packages/docker/build.sh#L127-L137)

**Comment by Denzy7** [comment]
Nice!
Screenshot_20240222-133100

**Review by TomJo2000 — APPROVED** [review]

File: root-packages/vnstat/src-common.h.patch
Comment (licy183) [inline root-packages/vnstat/src-common.h.patch:10]: You can replace `/data/data/com.termux/files/usr` with `@TERMUX_PREFIX@`.

**Comment by TomJo2000** [comment]
Looks good to me.
Thank you for your contribution,
I'd like to merge this once I get home, maybe 2-3 hours from now.

CC: @licy183 do you still have any outstanding concerns with this PR?

**Comment by TomJo2000** [comment]
Turns out we've had a package request open for this package for about 2 and a half years.
- #10143

**Review by licy183 — APPROVED** [review]
LGTM. Thanks for your contribution!

**Comment by TomJo2000** [comment]
I guess may as well merge this now then, wanted to hear back from licy before merging it.

**Pull request merged**
